hackathon_site/dashboard/frontend/src/components/teamDetail/TeamPendingOrderTable/TeamPendingOrderTable.tsx

```diff
@@ -0,0 +1,280 @@
+import {
+    Button,
+    Checkbox,
+    Grid,
+    Link,
+    Paper,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+} from "@material-ui/core";
+import { mockHardware, mockPendingOrdersInTable } from "testing/mockData";
+import React, { useState } from "react";
+import Container from "@material-ui/core/Container";
+import styles from "components/general/OrderTables/OrderTables.module.scss";
+import hardwareImagePlaceholder from "assets/images/placeholders/no-hardware-image.svg";
+import MenuItem from "@material-ui/core/MenuItem";
+import Select from "@material-ui/core/Select";
+import {
+    GeneralOrderTableTitle,
+    GeneralOrderTitle,
+} from "components/general/OrderTables/OrderTables";
+import { Formik } from "formik";
+
+const createDropdownList = (number: number) => {
+    let entry = [];
+
+    for (let i = 0; i <= number; i++) {
+        entry.push(
+            <MenuItem key={i} role="quantity" value={i.toString()}>
+                {i}
+            </MenuItem>
+        );
+    }
+
+    return entry;
+};
+
+const setInitialValues = (
+    request: { id: number; quantityRequested: number; quantityGranted: number }[]
+) => {
+    let orderInitialValues: Record<string, string | boolean> = {};
+    request.forEach((orderItem) => {
+        orderInitialValues[`${orderItem.id}-quantity`] =
+            orderItem.quantityGranted.toString();
+        orderInitialValues[`${orderItem.id}-checkbox`] = false;
+    });
+    return orderInitialValues;
+};
+
+export const TeamPendingOrderTable = () => {
+    const orders = mockPendingOrdersInTable;
+    const hardware = mockHardware;
+    const [visibility, setVisibility] = useState(false);
+    const toggleVisibility = () => {
+        // toggle visibility
```

Comment:
you can remove the comment because your code is very clear, no need to describe :)